p2p/listener.go

```diff
@@ -147,21 +159,26 @@ func (l *DefaultListener) listenRoutine() {
 	}
 }
 
-// A channel of inbound connections.
+// Connections returns a channel of inbound connections.
 // It gets closed when the listener closes.
 func (l *DefaultListener) Connections() <-chan net.Conn {
 	return l.connections
 }
 
+// InternalAddress returns the internal NetAddress (address used for
+// listening).
 func (l *DefaultListener) InternalAddress() *NetAddress {
 	return l.intAddr
 }
 
+// ExternalAddress returns the external NetAddress (publicly available,
+// determined using either UPnP or local resolver).
 func (l *DefaultListener) ExternalAddress() *NetAddress {
 	return l.extAddr
 }
 
-func (l *DefaultListener) ExternalAddressToString() string {
+// TODO: remove the need for this function by supporting IPv6 in NetAddress.
```

Comment:
actually, we may already support IPv6 in NetAddress. `net.IP` can hold either IPv4 or IPv6.